Tools/autotest/arduplane.py

```diff
@@ -4160,6 +4160,43 @@ def LandingDrift(self):
 
         self.wait_disarmed(timeout=180)
 
+    def CatapultTakeoff(self):
+        '''Test that a catapult takeoff works correctly'''
+
+        self.customise_SITL_commandline(
+            [],
+            model='plane-catapult',
+            defaults_filepath=self.model_defaults_filepath("plane")
+        )
+        self.set_parameters({
+            "TKOFF_THR_MINACC": 3.0,
+            "RTL_AUTOLAND": 2, # The mission contains a DO_LAND_START item.
+        })
+
+        # Record desired target speed.
+        min_speed = self.get_parameter("AIRSPEED_CRUISE")
+        max_throttle = self.get_parameter("TKOFF_THR_MAX")
+        if max_throttle == 0:
+            max_throttle = self.get_parameter("THR_MAX")
+
+        # Load and start mission.
+        self.load_mission("catapult.txt", strict=True)
+        self.change_mode('AUTO')
+
+        self.wait_ready_to_arm()
+        self.arm_vehicle()
+
+        # Throw the catapult.
```

Comment:
```suggestion
        self.progress("Throw the catapult")
```
Many times when I've put a comment like this in I've found it useful to put it in as a progress text instead.  Just a suggestion.

Reply:
Thanks! For now it's pretty evident (perhaps even more useful) by plotting RCOU[7]. But I'll keep that in mind.